sql/catalog.go

```diff
@@ -74,33 +74,33 @@ func (c *Catalog) AllDatabases() []Database {
 }
 
 // CreateDatabase creates a new Database and adds it to the catalog.
-func (c *Catalog) CreateDatabase(dbName string) error {
+func (c *Catalog) CreateDatabase(ctx *Context, dbName string) error {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
+	var err error
 	mut, ok := c.provider.(MutableDatabaseProvider)
 	if ok {
-		mut.CreateDatabase(dbName)
+		err = mut.CreateDatabase(ctx, dbName)
```

Comment:
Think this was better before, don't like the pattern of assigning an err in two branches and then returning it outside the conditional, has led to bugs in the past

Reply:
Just have both branches return